### PR TITLE
FOGL-8787: Fix for system failing due to delay in update of statistics and change of state of service

### PR DESCRIPTION
**Pull request by Mohit04tomar**

### CODE REVIEW

**Review by praveen-garg — DISMISSED** [review]

File: tests/system/python/packages/test_omf_north_service.py [hunk 1]
```diff
@@ -380,7 +380,7 @@ def test_omf_service_with_reconfig(self, reset_fledge, start_south_north, read_d
         put_url = "/fledge/category/{}".format(north_service_name)
         resp = utils.put_request(fledge_url, urllib.parse.quote(put_url), data)
         assert "Admin" == resp["PIWebAPIUserId"]["value"]
-
+        time.sleep(5)
```
Comment (ashish-jabble) [inline tests/system/python/packages/test_omf_north_service.py:383]: We should add comment here; Why do we need extra delay?

" As with JIRA number; delay in ...."

**Review by ashish-jabble — COMMENTED** [review]
Is this the only test which is failing? I suspect we have number of tests which are using to update statistics. Or may be with sometimes fails and pass.

**Review by praveen-garg — APPROVED** [review]

File: tests/system/python/api/test_statistics.py [hunk 2]
```diff
@@ -143,9 +143,8 @@ def test_statistics_history_with_service_enabled(self, start_south_coap, fledge_
         stats = utils.serialize_stats_map(jdoc)
         assert 1 == stats[ASSET_NAME.upper()]
         assert 1 == stats['READINGS']
-        
         # Allow stats collector schedule to run i.e. by default 15s
-        print("Waiting for 15 seconds for FOGL-8738")
+        print("Waiting for 16 seconds for delay caused by FOGL-8738 - north statistics update thread de-prioritised....")
```
Comment (ashish-jabble) [inline tests/system/python/api/test_statistics.py:147]: Do we really need to wait 16 seconds? How are you calculating this number? Are we good on Rpi? All supported engines verified?

File: tests/system/python/packages/test_omf_north_service.py [hunk 3]
```diff
@@ -259,7 +259,8 @@ def test_omf_service_with_enable_disable(self, reset_fledge, start_south_north,
         put_url = "/fledge/schedule/{}".format(north_schedule_id)
         resp = utils.put_request(fledge_url, urllib.parse.quote(put_url), data)
         assert False == resp['schedule']['enabled']
-
+        print("Waiting for 6 seconds for delay caused by FOGL-8813 - tune pre-fetch buffers...")
+        time.sleep(6)
```
Comment (praveen-garg) [inline tests/system/python/packages/test_omf_north_service.py:263]: Now this does not have flexibility to tune wait time per platform?

**Comment by Mohit04tomar** [comment]
I executed following test on the latest commit(787bab2):
1. http://jenkins.dianomic.com:8080/view/PI%20stuff/job/fledge_packages_system_test_north_service_ub1804/586/
2. http://jenkins.dianomic.com:8080/view/Fledge%20Pair%20tests/job/fledge_packages_pair_system_test_python_north_service_ub20x86_ub18aarch64/887/
3. http://jenkins.dianomic.com:8080/view/Fledge%20Pair%20tests/job/fledge_packages_pair_system_test_c_north_service_ub1804/810/

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**